gstd/gstd_debug.c

```diff
@@ -163,19 +179,30 @@ gstd_debug_get_property (GObject * object,
           self->threshold);
       g_value_set_string (value, self->threshold);
       break;
-
     case PROP_FLAGS:
       GST_DEBUG_OBJECT (self, "Returning flags %u", self->flags);
       g_value_set_flags (value, self->flags);
       break;
-
+    case PROP_RESET:
+      GST_WARNING_OBJECT (self, "Returning debug reset %d", self->reset);
+      g_value_set_boolean (value, self->reset);
+      break;
     default:
       /* We don't have any other property... */
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
       break;
   }
 }
 
+static void
+gstd_debug_set_threshold (GstdDebug * self)
+{
+  if (self->reset)
+    gst_debug_set_threshold_from_string ("*:0", TRUE);
```

Comment:
Please leave a comment explaining why is this needed.

Reply:
If reset is set, clear all the categories to 0. We run this manually
since the GStreamer gst_debug_set_threshold_from_string reset only **clears**
the categories that **weren't** set explicitly. Otherwise the threshold will
be appended to previously set thresholds